app/api/contact-history.js

```diff
@@ -10,17 +10,13 @@ async function getContactHistory (reference, logger) {
 
     return payload
   } catch (err) {
-    if (err.output.statusCode === 404) {
-      return null
-    } else {
-      logger.setBindings({ err, endpoint })
-      throw err
-    }
+    logger.setBindings({ err, endpoint })
+    throw err
   }
 }
 
 const getContactFieldData = (contactHistoryData, field) => {
-  const filteredData = contactHistoryData.filter(contact => contact?.data?.field === field)
+  const filteredData = contactHistoryData.filter(contact => contact.data?.field === field)
   if (filteredData.length) {
```

Comment:
Not your code I know, but could we swap this around so we have 

`if (!filteredData.length) return 'NA' `

And then we can have the happy path underneath, no need for an else?

Reply:
Go on then. Though I prefer positive assertions, so used `if (filteredData.length === 0)`